testdata/p4_16_errors_outputs/table-entries-optional-2-bmv2-frontend.p4

```diff
@@ -0,0 +1,80 @@
+#include <core.p4>
+#include <v1model.p4>
+
+header hdr {
+    bit<8>  e;
+    bit<16> t;
+    bit<8>  l;
+    bit<8>  r;
+    bit<8>  v;
+}
+
+struct Header_t {
+    hdr h;
+}
+
+struct Meta_t {
+}
+
+parser p(packet_in b, out Header_t h, inout Meta_t m, inout standard_metadata_t sm) {
+    state start {
+        b.extract<hdr>(h.h);
+        transition accept;
+    }
+}
+
+control vrfy(inout Header_t h, inout Meta_t m) {
+    apply {
+    }
+}
+
+control update(inout Header_t h, inout Meta_t m) {
+    apply {
+    }
+}
+
+control egress(inout Header_t h, inout Meta_t m, inout standard_metadata_t sm) {
+    apply {
+    }
+}
+
+control deparser(packet_out b, in Header_t h) {
+    apply {
+        b.emit<hdr>(h.h);
+    }
+}
+
+control ingress(inout Header_t h, inout Meta_t m, inout standard_metadata_t standard_meta) {
+    @name("ingress.a") action a() {
+        standard_meta.egress_spec = 9w0;
+    }
+    @name("ingress.a_with_control_params") action a_with_control_params(bit<9> x) {
+        standard_meta.egress_spec = x;
+    }
+    @name("ingress.t_optional") table t_optional_0 {
+        key = {
+            h.h.e: optional @name("h.h.e") ;
+            h.h.t: optional @name("h.h.t") ;
+        }
+        actions = {
+            a();
+            a_with_control_params();
+        }
+        default_action = a();
+        const entries = {
+                        (8w0xaa &&& 8w0xff, 16w0x1111 &&& 16w0xffff) : a_with_control_params(9w1);
+
+                        (8w0x0, default) : a_with_control_params(9w2);
```

Comment:
What does default mean in this context? I've looked through the P4 spec, but can't find a reference to it. (Likely there is nothing wrong here, I just wanted to understand).

Reply:
See section 8.12.2 "The universal set" in the latest P4_16 language spec.  You can use `default` or `_` in this context to mean "match anything", the same as 0 &&& 0 would mean for a ternary field.

Reply:
Ah, thank you, not sure how I missed that.